app/gui/src/project-view/util/codemirror/__tests__/markdownEditing.test.ts

```diff
@@ -0,0 +1,274 @@
+import { ensoMarkdown } from '@/components/MarkdownEditor/markdown'
+import {
+  HeaderLevel,
+  toggleHeader,
+  toggleList,
+  toggleQuote,
+} from '@/util/codemirror/markdownEditing'
+import { setVueHost } from '@/util/codemirror/vueHostExt'
+import { EditorState } from '@codemirror/state'
+import { EditorView } from '@codemirror/view'
+import { expect, test } from 'vitest'
+
+/**
+ * Setup editor with selection ranging from the first occurence of '|' in the `source` string to the last occurence of '|'.
+ * If there is a single '|', it points at the cursor position.
+ */
+const setupEditor = (source: string) => {
+  const selectionStart = source.indexOf('|')
+  const selectionEnd = source.lastIndexOf('|')
```

Comment:
This will be off by 1 because it is computed before the `replaceAll`.

Reply:
Nice catch, though it does not matter match in these tests. We are not too worried about tricky edge-cases, as even in the worse case edits are easily reversible even by hand.